scheduler/src/cook/kubernetes/api.clj

```diff
@@ -922,67 +922,70 @@
           ; * A job may have additional containers with the name aux-*
           job-status (first (filter (fn [c] (= cook-container-name-for-job (.getName c)))
                                     container-statuses))]
-      ; TODO: When we add logic here that supports detecting pods that have a i-am-being-preempted label, we need to modify
-      ; the controller to set the failure reason to unknown for pods in  :running,:missing.
       (if (some-> pod .getMetadata .getDeletionTimestamp)
         ; If a pod has been ordered deleted, treat it as if it was gone, It's being async removed.
         ; Note that we distinguish between this explicit :missing, and not being there at all when processing
         ; (:cook-expected-state/killed, :missing) in cook.kubernetes.controller/process
         {:state :missing
          :reason "Pod was explicitly deleted"
          :pod-deleted? true}
-        ; If pod isn't being async removed, then look at the containers inside it.
-        (if job-status
-          (let [^V1ContainerState state (.getState job-status)]
-            (cond
-              (.getWaiting state)
-              (if (pod-containers-not-initialized? (V1Pod->name pod) pod-status)
-                ; If the containers are not getting initialized,
-                ; then we should consider the pod failed. This
-                ; state can occur, for example, when volume
-                ; mounts fail.
-                {:state :pod/failed
-                 :reason "ContainersNotInitialized"}
-                {:state :pod/waiting
-                 :reason (-> state .getWaiting .getReason)})
-              (.getRunning state)
-              {:state :pod/running
-               :reason "Running"}
-              (.getTerminated state)
-              (let [exit-code (-> state .getTerminated .getExitCode)]
-                (if (= 0 exit-code)
-                  {:state :pod/succeeded
-                   :exit exit-code
-                   :reason (-> state .getTerminated .getReason)}
+        (if (some-> pod .getMetadata .getLabels (get (or (-> (config/kubernetes) :node-preempted-label) "node-preempted")))
+          {:state :missing
+           :reason "Node preempted"
+           :pod-deleted? true
+           :pod-preempted? true}
```

Comment:
Maybe we should use `cond` instead of nesting more `if`s? If we do, I think we could merge with the `cond` below into one big `cond`.

Reply:
The code below is "if job-status"
how would you want it to look?

Reply:
I see what you mean. I'm not sure. But seems like this function is getting out of hand. We don't have to fix it in this PR. Maybe extract functions or something?

Reply:
yeah